src/htmlrenderer.cpp

```diff
@@ -482,6 +490,102 @@ void HtmlRenderer::render(std::istream& input,
 				}
 				break;
 			}
+
+			case HtmlTag::VIDEO: {
+				std::string videourl;
+				try {
+					videourl = xpp.get_attribute_value("src");
+				} catch (const std::invalid_argument&) {
+					videourl = "";
+				}
+				video_count++;
+				if (videourl.length() > 0) {
+					link_num = add_link(links,
+							utils::censor_url(
+								utils::absolute_url(
+									url,
+									videourl)),
+							LinkType::VIDEO);
+					curline.append(strprintf::fmt(
+							"[%s %u (%s #%u)]",
+							_("video"),
+							video_count,
+							_("link"),
+							link_num));
+					source_count++;
```

Comment:
The contents of this `if` is repeated further down, in `case HtmlTag::SOURCE`. Would it be possible to move this into a function or a private method, and call that from both places? It can even be generalized to be used with `HtmlTag::VIDEO`, unless it hurts readability.

Reply:
I've added

Reply:
It looks excellent, thank you!